### PR TITLE
Standardize HDFS security section

### PR DESCRIPTION
**Pull request by NimaVaziri**

### CODE REVIEW

File: frameworks/hdfs/universe/config.json [hunk 1]
```diff
@@ -63,55 +63,59 @@
...
+        "security": {
+          "description": "HDFS security settings.",
           "type": "object",
-          "description": "TLS configuration properties.",
-          "properties": {
-            "enabled": {
-              "description": "Enable TLS support (requires Strict mode).",
-              "type": "boolean",
-              "default": false
+          "tls": {
```
Comment (elezar) [inline frameworks/hdfs/universe/config.json]: I minor nit. Could we move this to after the `kerberos` section like the other frameworks in #1871?

Alternatively, we can address the TLS move as a single PR there?

File: frameworks/hdfs/universe/config.json [hunk 2]
```diff
@@ -63,55 +63,59 @@
...
+            "type": "object",
+            "description": "TLS configuration properties.",
+            "properties": {
+              "enabled": {
+                "description": "Enable TLS support (requires Strict mode).",
```
Comment (elezar) [inline frameworks/hdfs/universe/config.json]: It was my understanding that this is an EE feature, but does work on permissive.

File: frameworks/hdfs/universe/marathon.json.mustache [hunk 3]
```diff
@@ -72,30 +72,30 @@
     "MESOS_HTTP_AUTHENTICATEE": "com_mesosphere_dcos_http_Authenticatee",
     {{/service.service_account_secret}}
 
-    {{#service.tls.enabled}}
-    "TLS_ENABLED": "{{service.tls.enabled}}",
-    "TASKCFG_ALL_TLS_ENABLED": "{{service.tls.enabled}}",
+    {{#service.security.tls.enabled}}
+    "TLS_ENABLED": "{{service.security.tls.enabled}}",
+    "TASKCFG_ALL_TLS_ENABLED": "{{service.security.tls.enabled}}",
```
Comment (elezar) [inline frameworks/hdfs/universe/marathon.json.mustache]: out of scope, but are both environment variables required?

File: frameworks/hdfs/universe/marathon.json.mustache [hunk 4]
```diff
@@ -72,30 +72,30 @@
...
-    "TASKCFG_ALL_KDC_HOST_NAME": "{{service.kerberos.kdc_host_name}}",
-    "TASKCFG_ALL_KDC_HOST_PORT": "{{service.kerberos.kdc_host_port}}",
-    "KEYTAB_SECRET_PATH": "{{service.kerberos.keytab_secret}}",
-    "TASKCFG_ALL_KERBEROS_PRIMARY": "{{service.kerberos.primary}}",
-    "TASKCFG_ALL_KERBEROS_PRIMARY_HTTP": "{{service.kerberos.primary_http}}",
-    "TASKCFG_ALL_KERBEROS_REALM": "{{service.kerberos.realm}}",
-    {{/service.kerberos.enabled}}
+    {{#service.security.kerberos.enabled}}
+    "KERBEROS_ENABLED": "{{service.security.kerberos.enabled}}",
+    "TASKCFG_ALL_KERBEROS_ENABLED": "{{service.security.kerberos.enabled}}",
```
Comment (elezar) [inline frameworks/hdfs/universe/marathon.json.mustache]: Same question regarding the environment variables?

File: frameworks/hdfs/universe/marathon.json.mustache [hunk 5]
```diff
@@ -72,30 +72,30 @@
...
+    "TASKCFG_ALL_KDC_HOST_NAME": "{{service.security.kerberos.kdc_host_name}}",
+    "TASKCFG_ALL_KDC_HOST_PORT": "{{service.security.kerberos.kdc_host_port}}",
+    "KEYTAB_SECRET_PATH": "{{service.security.kerberos.keytab_secret}}",
```
Comment (elezar) [inline frameworks/hdfs/universe/marathon.json.mustache]: Minor nit. Could we at least rename this environment variable to remove the `_PATH`? Ideally, we could have the same environment variables as the other services, but I can address this as part of #1871.

File: frameworks/hdfs/universe/config.json (hunk 1 above)
Comment (NimaVaziri) [reply frameworks/hdfs/universe/config.json]: Done

File: frameworks/hdfs/universe/config.json (hunk 2 above)
Comment (NimaVaziri) [reply frameworks/hdfs/universe/config.json]: Yeah, possibly stale documentation, removed strict wording.

File: frameworks/hdfs/universe/marathon.json.mustache (hunk 3 above)
Comment (NimaVaziri) [reply frameworks/hdfs/universe/marathon.json.mustache]: See discussion here: https://github.com/mesosphere/dcos-commons/pull/1789/files/e1c17f46f16f407a8276af8ff439821eb1b2c47f#r148108353

**Comment by NimaVaziri** [comment]
Will convert env var notation to `SECURITY_` to ensure consistency with https://github.com/mesosphere/dcos-commons/pull/1871.

File: frameworks/hdfs/src/main/dist/svc.yml [hunk 6]
```diff
@@ -57,11 +57,11 @@ pods:
               ./bootstrap && ./{{HDFS_VERSION}}/bin/hdfs journalnode
         resource-set: journal-resources
         configs:
-          {{#KERBEROS_ENABLED}}
+          {{#SECURITY_KERBEROS_ENABLED}}
           krb5-conf:
             template: krb5.conf
             dest: {{JRE_VERSION}}/lib/security/krb5.conf
```
Comment (elezar) [inline frameworks/hdfs/src/main/dist/svc.yml]: Is it *required* that this be copied to `{{JRE_VERSION}}/lib/security/krb5.conf`? This breaks as soon as we update the JRE version. Would it not be more scalable to use `{{HDFS_VERSION}}/etc/security/krb5.conf` here?

File: frameworks/hdfs/src/main/dist/svc.yml [hunk 7]
```diff
@@ -158,19 +158,19 @@ pods:
         readiness-check:
           cmd: >
             export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/) &&
-          {{#KERBEROS_ENABLED}}
-            KRB5_CONFIG={{JRE_VERSION}}/lib/security/krb5.conf kinit -k -t $MESOS_SANDBOX/hdfs.keytab $KERBEROS_PRIMARY/$TASK_NAME.$FRAMEWORK_HOST@$KERBEROS_REALM &&
-          {{/KERBEROS_ENABLED}}
+          {{#SECURITY_KERBEROS_ENABLED}}
+            KRB5_CONFIG={{JRE_VERSION}}/lib/security/krb5.conf kinit -k -t $MESOS_SANDBOX/hdfs.keytab $SECURITY_KERBEROS_PRIMARY/$TASK_NAME.$FRAMEWORK_HOST@$SECURITY_KERBEROS_REALM &&
```
Comment (elezar) [inline frameworks/hdfs/src/main/dist/svc.yml]: Just a question for myself. Does HDFS not support JAAS, or does it require that `kinit` be used explicitly.

Comment (NimaVaziri) [reply frameworks/hdfs/src/main/dist/svc.yml]: It actually doesn't require it, as JAAS is also supported.

File: frameworks/hdfs/src/main/dist/svc.yml (hunk 6 above)
Comment (NimaVaziri) [reply frameworks/hdfs/src/main/dist/svc.yml]: There's already a JIRA for this, so I might as well just fix it here.

Comment (elezar) [reply frameworks/hdfs/src/main/dist/svc.yml]: Sure, let's not do it here. The other ticket will be fine.

**Comment by elezar** [comment]
@NimaVaziri I am closing this PR and continuing work in #1871.

**Pull request closed without merging**